make-release.sh

```diff
@@ -0,0 +1,126 @@
+      # create pull request for master branch, as branch is restricted
+      git branch "${PR_BRANCH}"
+      git checkout "${PR_BRANCH}"
+      git pull origin "${PR_BRANCH}"
+      git push origin "${PR_BRANCH}"
+      lastCommitComment="$(git log -1 --pretty=%B)"
+      hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
+    fi 
+  fi
+  git checkout ${CURRENT_BRANCH}
+}
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [--trigger-release]"
+  echo "Example: $0 --repo git@github.com:eclipse/che-subproject --version 7.7.0 --trigger-release"; echo
+}
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]]; then
+  usage
+  exit 1
+fi
+
+QUAY_REPO="quay.io/eclipse/che-workspace-loader:${VERSION}"
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# if doing a .0 release, use master; if doing a .z release, use $BRANCH
+if [[ ${VERSION} == *".0" ]]; then
+  BASEBRANCH="master"
+else 
+  BASEBRANCH="${BRANCH}"
+fi
+
+# get sources from ${BASEBRANCH} branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}"
+git checkout "${BASEBRANCH}"
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+fi
+
+set -e
+
+# change VERSION file
+npm --no-git-tag-version version --allow-same-version ${VERSION}
+
+# commit change into branch
+if [[ ${NOCOMMIT} -eq 0 ]]; then
+  COMMIT_MSG="[release] Bump to ${VERSION} in ${BRANCH}"
+  git commit -asm "${COMMIT_MSG}"
+  git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+fi
+
+if [[ $TRIGGER_RELEASE -eq 1 ]]; then
+  # push new branch to release branch to trigger CI build
+  git checkout "${BRANCH}"
+  docker build -t "${QUAY_REPO}" -f apache.Dockerfile .
+  docker push "${QUAY_REPO}"
+
+  # tag the release
+  git checkout "${BRANCH}"
+  git tag "${VERSION}"
+  git push origin "${VERSION}"
+fi
+
+# now update ${BASEBRANCH} to the new snapshot version
+git checkout "${BASEBRANCH}"
+
+# change VERSION file + commit change into ${BASEBRANCH} branch
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  # bump the y digit, if it is a major release
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.10.x, get BASE=7, NEXT=11
+  NEXT_VERSION_Y="${BASE}.${NEXT}.0-SNAPSHOT"
+  bump_version ${NEXT_VERSION_Y} ${BASEBRANCH}
+fi
+# bump the z digit
+[[ $VERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.7.1, get BASE=7.7, NEXT=2
+NEXT_VERSION_Z="${BASE}.${NEXT}-SNAPSHOT"
+bump_version ${NEXT_VERSION_Z} ${BRANCH}
```

Comment:
![image](https://user-images.githubusercontent.com/436777/98543967-93320d80-2293-11eb-9ac1-54cd422d0f53.png)

it doesn't pass shellcheck.net